packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts

```diff
@@ -203,6 +208,27 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumerBase {
             };
         }
 
+        if (isDirectSwapCompatible(quote, optsWithDefaults, [ERC20BridgeSource.Mooniswap])) {
+            const fillData = quote.orders[0].fills[0].fillData as MooniswapFillData;
+            return {
+                calldataHexString: this._exchangeProxy
+                    .sellToLiquidityProvider(
+                        isFromETH ? ETH_TOKEN_ADDRESS : sellToken,
+                        isToETH ? ETH_TOKEN_ADDRESS : buyToken,
+                        MOONISWAP_LIQUIDITY_PROVIDER_BY_CHAIN_ID[this.chainId],
+                        NULL_ADDRESS,
+                        sellAmount,
+                        minBuyAmount,
+                        poolEncoder.encode([fillData.poolAddress]),
+                    )
+                    .getABIEncodedTransactionData(),
+                ethAmount: isFromETH ? sellAmount : ZERO_AMOUNT,
+                toAddress: this._exchangeProxy.address,
+                allowanceTarget: this.contractAddresses.exchangeProxy,
+                gasOverhead: ZERO_AMOUNT,
```

Comment:
It definitely isn't zero (same for curve LP). Need to figure this out.

Reply:
Might just guess something like 25k for now.

Reply:
I think ZERO is fine. What we were really trying to get at is any overhead introduced in encoding that already isn't accounted for in just the pure route cost.

Why you think 25k for this?

For example adding a `PositiveSlippageFee` has a hidden future cost so it sets `gasOverhead` which gets added on to any `eth_estimateGas`. Perhaps the naming can be better.

Reply:
These LPs technically add a hop because we still move taker funds into them then they transact with curve/mooni. They're basically bridges.

Reply:
oooooooooook I get it now lol. Back to zero.